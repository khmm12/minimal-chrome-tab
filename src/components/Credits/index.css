.aza {
  align-self: flex-end;
  text-align: right;
  font-size: 1.6rem;
  font-style: italic;
  text-transform: uppercase;
}
