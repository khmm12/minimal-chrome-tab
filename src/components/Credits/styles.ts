import { css } from 'styled-system/css'

export const container = css.raw({
  textAlign: 'right',
  fontSize: '1rem',
})

export const link = css.raw({
  display: 'inline-block',
  textDecoration: 'none',
  fontStyle: 'italic',
  textTransform: 'uppercase',
  color: { base: 'neutral.700', _dark: 'neutral.300', _hover: { base: 'black', _dark: 'white' } },
  transform: { base: 'scale(1)', _hover: 'scale(1.05)' },
  transition: '0.1s transform ease-out, 0.1s color ease-out',
})
