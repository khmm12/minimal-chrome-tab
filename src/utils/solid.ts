export type SignalValue<T> = () => T

export function unwrap<T>(mayBeSingnal: SignalValue<T> | Exclude<T, Function>): T {
  // @ts-expect-error
  return typeof mayBeSingnal === 'function' ? mayBeSingnal() : mayBeSingnal
}
