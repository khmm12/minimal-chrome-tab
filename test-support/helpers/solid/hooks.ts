import { createRoot } from 'solid-js'

type Disposal = () => void

const disposals = new Set<Disposal>()

afterEach(() => cleanup())

export function renderHook<T>(fn: () => T): T {
  return withRoot(fn)[0]
}

export function withRoot<T>(fn: () => T): [value: T, dispose: () => void] {
  let disposal: () => void

  const value = createRoot((dispose) => {
    disposal = dispose
    disposals.add(dispose)
    return fn()
  })

  const dispose = (): void => {
    disposal()
    disposals.delete(disposal)
  }

  return [value, dispose]
}

export function cleanup(): void {
  for (const dispose of disposals) dispose()
  disposals.clear()
}
