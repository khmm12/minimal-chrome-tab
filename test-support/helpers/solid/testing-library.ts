import type { JSX } from 'solid-js'
import { hydrate as solidHydrate, render as solidRender } from 'solid-js/web'
import { getQueriesForElement, prettyDOM, queries as allQueries } from '@testing-library/dom'
import type { Queries, BoundFunction, prettyFormat } from '@testing-library/dom'

interface Ref {
  container: HTMLElement
  unmount: () => void
  dispose: () => void
}

type Ui = () => JSX.Element

interface Options<Q extends Queries = typeof allQueries> {
  container?: HTMLElement
  baseElement?: HTMLElement
  queries?: Q
  hydrate?: boolean
}

type DebugFn = (
  baseElement?: HTMLElement | HTMLElement[],
  maxLength?: number,
  options?: prettyFormat.OptionsReceived
) => void

type Result<Q extends Queries = typeof allQueries> = {
  container: HTMLElement
  baseElement: HTMLElement
  debug: DebugFn
  unmount: () => void
} & { [P in keyof Q]: BoundFunction<Q[P]> }

afterEach(() => cleanup())

const mountedContainers = new Set<Ref>()

function render<Q extends Queries = typeof allQueries>(ui: Ui, options: Options<Q>): Result<Q>
function render(ui: Ui, options?: Omit<Options, 'queries'>): Result

function render(ui: Ui, options: Options = {}): unknown {
  let { container, baseElement = container ?? document.body, queries, hydrate = false } = options

  container ??= baseElement.appendChild(document.createElement('div'))

  const unmount = (): void => {
    if (container?.parentElement === document.body) container.remove()
  }

  const dispose = hydrate ? solidHydrate(ui, container) : solidRender(ui, container)

  mountedContainers.add({ container, unmount, dispose })

  const debug: DebugFn = (el = baseElement, maxLength, options) =>
    Array.isArray(el)
      ? el.forEach((e) => console.log(prettyDOM(e, maxLength, options)))
      : console.log(prettyDOM(el, maxLength, options))

  return {
    container,
    baseElement,
    debug,
    unmount: dispose,
    ...getQueriesForElement(baseElement, queries),
  }
}

function cleanup(): void {
  mountedContainers.forEach(cleanupAtContainer)
  mountedContainers.clear()
}

function cleanupAtContainer(ref: Ref): void {
  ref.dispose()
  ref.unmount()
  mountedContainers.delete(ref)
}

export { render, cleanup }
