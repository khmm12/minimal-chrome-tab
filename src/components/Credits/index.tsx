import type { JSX } from 'solid-js'
import { css } from 'styled-system/css'
import * as s from './styles'

const link = 'https://github.com/khmm12/minimal-chrome-tab'

export default function Credits(): JSX.Element {
  return (
    <div class={css(s.container)}>
      <a class={css(s.link)} href={link} target="_blank" rel="noreferrer">
        Made by khmm12
      </a>
    </div>
  )
}
