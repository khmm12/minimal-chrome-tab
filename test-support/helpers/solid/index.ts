import { render, cleanup as cleanupElements } from './testing-library'
import { renderHook, withRoot, cleanup as cleanupHooks } from './hooks'

export { renderHook, withRoot, render }

export function cleanup(): void {
  cleanupHooks()
  cleanupElements()
}

export * from '@testing-library/dom'
