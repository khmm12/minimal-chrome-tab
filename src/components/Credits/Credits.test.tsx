import { render, screen } from '@test/helpers/solid'
import Credits from '.'

describe('Credits', () => {
  it('renders correctly', async () => {
    render(() => <Credits />)

    const $credits = screen.getByText('Made by khmm12')

    expect($credits).toBeInTheDocument()
    expect($credits).toMatchSnapshot()
  })
})
