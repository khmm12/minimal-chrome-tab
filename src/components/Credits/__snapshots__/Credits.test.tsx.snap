// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Credits > renders correctly 1`] = `
<a
  class="d_inline-block td_none font-style_italic tt_uppercase c_slate.700 dark:c_slate.300 hover:c_black hover:dark:c_white trf_scale(1) hover:trf_scale(1.05) trs_0.1s_transform_ease-out,_0.1s_color_ease-out"
  href="https://github.com/khmm12/minimal-chrome-tab"
  rel="noreferrer"
  target="_blank"
>
  Made by khmm12
</a>
`;
